Feature/cron

viacheslavbocharov opened the pull request:
feat: add fastify-cron plugin to refresh RSS feeds every 10 minutes

viacheslavbocharov committed:
feat: add fastify-cron plugin to refresh RSS feeds every 10 minutes

viacheslavbocharov committed:
feat: biome linting

Gunnar97 commented on src/plugins/cron-feed.plugin.ts:
```diff
@@ -0,0 +1,78 @@
+import { parseFeed } from "../modules/feedParser/services/feedParserService";
+import { bulkUpsertFeedItems, upsertFeedCache } from "../services/dbService";
+
+export default fp(async function cronFeedPlugin(app: FastifyInstance) {
```
у тебя в апп разве const app =

viacheslavbocharov replied:
Fixed

Gunnar97 commented on src/plugins/cron-feed.plugin.ts:
```diff
@@ -0,0 +1,78 @@
+
+          for (const url of urls) {
+            try {
+              const { title, items } = await parseFeed(url);
```
в кроне должен біть только візов сервиса вся логика в сервисе

viacheslavbocharov replied:
Fixwd

Gunnar97 commented on src/plugins/cron-feed.plugin.ts:
```diff
@@ -0,0 +1,78 @@
+              }));
+
+              await upsertFeedCache(server.prisma, {
+                url,
```
откдв взялся server

viacheslavbocharov replied:
Fixed

viacheslavbocharov committed:
fix(cron-feed): replace undefined server reference with fastify instance

Use fastify inside onTick instead of server to keep consistency and avoid undefined variable.

viacheslavbocharov committed:
fix(cron-feed): biome linting
